tensorboard/webapp/hparams/_redux/hparams_selectors.ts

```diff
@@ -25,81 +30,126 @@ import {
-  getHparamsState,
-  (
-    state: HparamsState,
-    experimentIds: string[]
-  ): Map<string, DiscreteFilter | IntervalFilter> => {
-    const defaultFilterMaps: Array<
-      Map<string, DiscreteFilter | IntervalFilter>
-    > = [];
-
-    for (const experimentId of experimentIds) {
-      if (!state.specs[experimentId]) {
-        continue;
-      }
-
-      defaultFilterMaps.push(state.specs[experimentId].hparam.defaultFilters);
+function getHparamsDefaultFiltersForExperimentsResultFunction(
+  state: HparamsState,
+  experimentIds: string[]
+): Map<string, DiscreteFilter | IntervalFilter> {
+  const defaultFilterMaps: Array<Map<string, DiscreteFilter | IntervalFilter>> =
+    [];
+
+  for (const experimentId of experimentIds) {
+    if (!state.specs[experimentId]) {
+      continue;
     }
 
-    return combineDefaultHparamFilters(defaultFilterMaps);
+    defaultFilterMaps.push(state.specs[experimentId].hparam.defaultFilters);
   }
-);
 
-export const getHparamFilterMap = createSelector(
-  getHparamsDefaultFiltersForExperiments,
+  return combineDefaultHparamFilters(defaultFilterMaps);
+}
+
+const getHparamsDefaultFiltersForExperiments = createSelector(
   getHparamsState,
-  (
-    combinedDefaultfilterMap,
-    hparamState,
-    experimentIds: string[]
-  ): Map<string, IntervalFilter | DiscreteFilter> => {
-    const id = getIdFromExperimentIds(experimentIds);
-    const otherFilter = hparamState.filters[id];
-
-    return new Map([
-      ...combinedDefaultfilterMap,
-      ...(otherFilter?.hparams ?? []),
-    ]);
-  }
+  getHparamsDefaultFiltersForExperimentsResultFunction
 );
 
-const getMetricsDefaultFiltersForExperiments = createSelector(
-  getHparamsState,
-  (
-    state: HparamsState,
-    experimentIds: string[]
-  ): Map<string, IntervalFilter> => {
-    const defaultFilterMaps: Array<Map<string, IntervalFilter>> = [];
+function getHparamsDefaultFiltersForExperimentsFromExperimentIds(
+  experimentIds: string[]
+) {
+  return createSelector(getHparamsState, (state) =>
+    getHparamsDefaultFiltersForExperimentsResultFunction(state, experimentIds)
+  );
+}
+
+function getHparamFilterMapResultFunction(
+  hparamState: HparamsState,
+  combinedDefaultfilterMap: Map<string, DiscreteFilter | IntervalFilter>,
+  experimentIds: string[]
+): Map<string, IntervalFilter | DiscreteFilter> {
+  const id = getIdFromExperimentIds(experimentIds);
+  const otherFilter = hparamState.filters[id];
+
+  return new Map([
+    ...combinedDefaultfilterMap,
+    ...(otherFilter?.hparams ?? []),
+  ]);
+}
 
-    for (const experimentId of experimentIds) {
-      if (!state.specs[experimentId]) {
-        continue;
-      }
+export const getHparamFilterMap = createSelector(
+  getHparamsState,
+  getHparamsDefaultFiltersForExperiments,
+  getHparamFilterMapResultFunction
+);
 
-      defaultFilterMaps.push(state.specs[experimentId].metric.defaultFilters);
+export function getHparamFilterMapFromExperimentIds(experimentIds: string[]) {
```

Comment:
This is the one we want to use in the future?

Reply:
Yes